src/backend/trace.rs

```diff
@@ -381,3 +433,240 @@ impl TraceData {
+                ttl: value.ttl(),
+                round: value.round(),
+                sent: Some(value.sent()),
+                status: value.status(),
+                host: value.host(),
+                received: value.received(),
+                icmp_packet_type: Some(IcmpPacketType::NotApplicable),
+                extensions: None,
+            }
+        }
+    }
+    #[derive(Deserialize,Serialize,Debug,Clone)]
+    struct HopResults {
+        /// The total probes sent for this hop.
+        total_sent: usize,
+        /// The total probes received for this hop.
+        total_recv: usize,
+        /// The round trip time for this hop in the current round.
+        last: Option<Duration>,
+        /// The best round trip time for this hop across all rounds.
+        best: Option<Duration>,
+        /// The worst round trip time for this hop across all rounds.
+        worst: Option<Duration>,
+        /// The history of round trip times across the last N rounds.
+        mean: f64,
+        m2: f64,
+        /// The ABS(RTTx - RTTx-n)
+        jitter: Option<Duration>,
+        /// The Sequential jitter average calculated for each
+        javg: Option<f64>,
+        /// The worst jitter reading recorded
+        jmax: Option<Duration>,
+        /// The interval calculation i.e smooth
+        jinta: f64,
+    }
+    impl HopResults {
+        #[allow(clippy::too_many_arguments)]
+        fn new(
+            total_sent: usize,
+            total_recv: usize,
+            last: u64,
+            best: u64,
+            worst: u64,
+            mean: f64,
+            m2: f64,
+            jitter: u64,
+            javg: f64,
+            jmax: u64,
+            jinta: f64,
+        ) -> Self {
+                Self {total_sent,
+                total_recv,
+                last: Some(Duration::from_millis(last)),
+                best: Some(Duration::from_millis(best)),
+                worst: Some(Duration::from_millis(worst)),
+                mean,
+                m2,
+                jitter: Some(Duration::from_millis(jitter)),
+                javg: Some(javg),
+                jmax: Some(Duration::from_millis(jmax)),
+                jinta,
+            }
+        }
+        //TODO: Create combined struct ImportProbe Vec<ImportProbe> & HopResults.
+        //JSON will define the test and expected results.
+    }
+    impl PartialEq<&Hop> for HopResults {
+        fn eq(&self, other: &&Hop) -> bool {
+            self.last == other.last
+            && self.jitter == other.jitter
+            && self.jmax == other.jmax
+            && self.javg == other.javg
+            && format!("{:.2}",self.jinta) == format!("{:.2}", other.jinta)
+            }
+    }    
+    #[test_case("base_line.json", &HopResults::new(2,2,700,300,700,0.0,0.0,400,350.0,400,680.28))]
+    fn test_probe_file(json_file: &str, expected: &HopResults) {
+        let mut json = String::new();
+        let mut dir_json_file = "./tests/data/".to_owned();
```

Comment:
Can we use include_str!() to read these at compile time?  See https://github.com/fujiapple852/trippy/blob/55641c6a60bde36a3a08f72f15b6dfa49215d994/tests/sim/tests.rs#L31 for an example